dftimewolf/lib/collectors/bigquery.py

```diff
@@ -25,36 +25,37 @@ def __init__(self,
     self._project_name = ""
     self._query = ""
     self._description = ""
+    self._panda_output = False
 
   # pylint: disable=arguments-differ
-  def SetUp(self, project_name: str, query: str, description: str) -> None:
+  def SetUp(self,
+            project_name: str,
+            query: str,
+            description: str,
+            panda_output: bool) -> None:
     """Sets up a BigQuery collector.
 
     Args:
       project_name (str): name of the project that contains the BigQuery tables.
       query (str): The query to run.
       description (str): A description of the query.
+      panda_output (bool): True if the results should be kep in a panda DF in
+          memory, false if the they should be written to disk.
     """
     self._project_name = project_name
     self._query = query
     self._description = description
+    self._panda_output = panda_output
 
   def Process(self) -> None:
     """Collects data from BigQuery."""
-    output_file = tempfile.NamedTemporaryFile(
-        mode='w', delete=False, encoding='utf-8', suffix='.jsonl')
-    output_path = output_file.name
-    self.logger.info(f'Downloading results to {output_path:s}')
 
     try:
       if self._project_name:
         bq_client = bigquery.Client(project=self._project_name)
       else:
         bq_client = bigquery.Client()
-
-      records = bq_client.query(self._query).to_dataframe().to_json(
-          orient='records', lines=True, date_format='iso')
-      output_file.write(records)
+      df = bq_client.query(self._query).to_dataframe()
```

Comment:
Not familiar with the API but does the NotFound / DefaultCredentialsError exceptions apply to the query() call here?

Reply:
DefaultCredentialsError is raised by the the creation of the client object in lines 55/57, NotFound appears to come from methods used by `query`. 

https://github.com/googleapis/python-bigquery/blob/main/google/cloud/bigquery/client.py#L188

Reply:
thanks for clarifying